docs - reorganize/reorder the reference guide topics

lisakowen opened the pull request:
reorganize and reorder the reference guide topics:
- locate more end-user-focused topics near the top  
- group some topics together, add landing pages
- misc edits to PL titles to remove term "extension"
- remove some unused files
- also add a few items to main left subnav

doc review site link:  http://docs-lisa-refreorg.cfapps.io/6-0/main/index.html

dyozie commented on gpdb-doc/dita/ref_guide/extensions/extensions.ditamap:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">
+<map title="Procedural Language, Maching Learning, and Geospatial Extensions">
```
Maching -> Machine

dyozie commented on gpdb-doc/dita/ref_guide/misc.ditamap:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">
+<map title="Features, Keywords, and Compliance">
+  <topicref href="misc.xml" navtitle="SQL Features, Reserved Key Words, and Compliance in Greenplum">
```
As an aside, it's easier to change this in the referenced file (misc.xml) and then omit the navtitle= entry, as it will be pulled in automatically.

dyozie commented on gpdb-doc/dita/ref_guide/ref_guide.ditamap:
```diff
@@ -163,10 +163,19 @@
-		<topicref href="env_var_ref.xml" navtitle="Greenplum Environment Variables" id="env_var_ref"/>
-		<topicref href="sql-keywords.xml" navtitle="Reserved Identifiers and SQL Keywords" id="sql_keywords"/>
-		<topicref href="system_catalogs/catalog_ref.xml#topic1">
+		<topicref href="data_types.xml" navtitle="Greenplum Database Data Types" id="data_types">
```
Let's remove "Greenplum Database" from the navtitle here -> it's generally easier to scan these things if everything doesn't use the Greenplum prefix.

dyozie replied:
(and do the same with the other high-level topics)

lisakowen replied:
removed the "Greenplum" prefix from most subnav and topic titles.  kept it in a few places (GPPC, mapreduce).

lisakowen commented on gpdb-doc/dita/ref_guide/extensions/extensions.ditamap:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">
+<map title="Procedural Language, Maching Learning, and Geospatial Extensions">
```
this title is now equivalent to the topic title, "Procedural Languages and Analytics Extensions".

lisakowen commented on gpdb-doc/dita/ref_guide/extensions/extensions.ditamap:
```diff
@@ -0,0 +1,14 @@
+    <topicref href="pl_python.xml" linking="none"/>
+    <topicref href="pl_r.xml" linking="none"/>
+    <topicref href="pl_sql.xml" linking="none"/>
+    <topicref href="madlib.xml" linking="none"/>
```
the landing page is grouped by PLs and analytics extensions.

lisakowen commented:
i think that i incorporated all of your requested edits, @dyozie .  pushed changes to the review site.  (may need to refresh.)

litzell commented on gpdb-doc/dita/ref_guide/ref_guide.ditamap:
```diff
@@ -163,10 +163,18 @@
+		<topicref href="system_catalogs/catalog_ref.xml">
 			<topicref href="system_catalogs/catalog_ref-tables.xml#topic1"/>
 			<topicref href="system_catalogs/catalog_ref-views.xml#topic1"/>
 			<topicref href="system_catalogs/catalog_ref-html.xml#topic1">
```
We have an alpha list of tables, an alpha list of views, and an alpha list of tables and views with the refs nested beneath. I'm wondering if we really need the combined list. If the table/view refs were moved up a level, it would be a flatter nav like the SQL Command Reference.

lisakowen commented:
@litzell - yes, the system catalog section needs some work.  would like to tackle that in a separate PR.

lisakowen commented:
thanks for reviewing @litzell .  i updated the sql commands and system catalog subnav entries and topic titles.  good suggestion!

also updated the review site, refresh ...
